### PR TITLE
enhance: wal adaptor implementation

### PR DESCRIPTION
**Pull request by chyezh**
issue: #33285

- add adaptor to implement walimpls into wal interface.
- implement timetick sorted and filtering scanner.
- add test for wal.

### CODE REVIEW

**Comment by chyezh** [comment]
rerun ut

File: internal/streamingnode/server/wal/adaptor/scanner_adaptor.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+func (s *scannerAdaptorImpl) getEventCh(scanner walimpls.ScannerImpls) (<-chan message.ImmutableMessage, chan<- message.ImmutableMessage) {
+	if s.pendingQueue.Len() == 0 {
+		// If pending queue is empty,
+		// no more message can be sent,
+		// we always need to recv message from upstream to avoid starve.
+		return scanner.Chan(), nil
+	}
+	// TODO: configurable pending count.
+	if s.pendingQueue.Len()+s.reorderBuffer.Len() > 1024 {
```
Comment (chyezh) [inline internal/streamingnode/server/wal/adaptor/scanner_adaptor.go:100]: here, if our timetick interval is greater than 1024, the scanner get stucked.
Should be modified into another strategy.

File: internal/streamingnode/server/wal/interceptors/timetick/timetick_interceptor.go [hunk 2]
```diff
@@ -65,6 +66,37 @@ func (impl *timeTickAppendInterceptor) executeSyncTimeTick(interval time.Duratio
...
+		case <-ticker.C:
+			if err := impl.sendTsMsg(impl.ctx, wal.Append); err != nil {
+				log.Warn("send time tick sync message failed", zap.Error(err))
+			}
+		}
+	}
+}
+
+// blockUntilSyncTimeTickReady blocks until the first time tick message is sent.
+func (impl *timeTickAppendInterceptor) blockUntilSyncTimeTickReady(underlyingWALImpls walimpls.WALImpls) error {
```
Comment (chyezh) [inline internal/streamingnode/server/wal/interceptors/timetick/timetick_interceptor.go:95]: move first timetick sent as a single function.

**Comment by jaime0815** [comment]
All refinements or fixes will be done in the next PR.

**Pull request merged**